domain/infosync/info.go

```diff
@@ -423,8 +423,19 @@ func GetRuleBundle(ctx context.Context, name string) (*placement.Bundle, error)
 	return bundle, err
 }
 
-// PutRuleBundle is used to post one specific rule bundle to PD.
-func PutRuleBundle(ctx context.Context, bundle *placement.Bundle) error {
+// PutRuleBundles is used to post specific rule bundles to PD.
+func PutRuleBundles(ctx context.Context, bundles []*placement.Bundle) error {
+	empty := true
+	for _, bundle := range bundles {
+		if len(bundle.Rules) != 0 || bundle.Index != 0 || bundle.Override != false {
+			empty = false
+			break
+		}
+	}
+	if empty {
+		return nil
+	}
```

Comment:
This is checking whether the bundle to push is empty, not that the original bundle of this partition is empty.

Reply:
For example, there exists a rule for one partition, to delete the rule for this partition, a bundle is still needed to push to PD.